x/staking/src/keeper/store_iter.rs

```diff
@@ -0,0 +1,63 @@
+use super::Database;
+use gears::{
+    store::database::{ext::UnwrapCorrupt, prefix::PrefixDB},
+    types::{
+        address::ValAddress,
+        store::{gas::errors::GasStoreErrors, kv::Store, range::StoreRange},
+    },
+};
+use std::{borrow::Cow, ops::Bound};
+
+#[derive(Debug)]
+struct StoreIterator<'a, DB>(StoreRange<'a, PrefixDB<DB>>);
```

Comment:
Maybe we need this structure in gears?

Reply:
We can have such structs. One for infallible stores, the other one for fallible.

Reply:
If you want